tests/components/yale/test_binary_sensor.py

```diff
@@ -0,0 +1,390 @@
+"""The binary_sensor tests for the yale platform."""
+
+import datetime
+from unittest.mock import patch
+
+from homeassistant.components.lock import DOMAIN as LOCK_DOMAIN
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    SERVICE_LOCK,
+    SERVICE_UNLOCK,
+    STATE_OFF,
+    STATE_ON,
+    STATE_UNAVAILABLE,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import device_registry as dr
+import homeassistant.util.dt as dt_util
+
+from .mocks import (
+    _create_yale_with_devices,
+    _mock_activities_from_fixture,
+    _mock_doorbell_from_fixture,
+    _mock_doorsense_enabled_yale_lock_detail,
+    _mock_lock_from_fixture,
+)
+
+from tests.common import async_fire_time_changed
+
+
+async def test_doorsense(hass: HomeAssistant) -> None:
+    """Test creation of a lock with doorsense and bridge."""
+    lock_one = await _mock_lock_from_fixture(
+        hass, "get_lock.online_with_doorsense.json"
+    )
+    await _create_yale_with_devices(hass, [lock_one])
+
+    binary_sensor_online_with_doorsense_name = hass.states.get(
+        "binary_sensor.online_with_doorsense_name_door"
+    )
+    assert binary_sensor_online_with_doorsense_name.state == STATE_ON
+
+    data = {ATTR_ENTITY_ID: "lock.online_with_doorsense_name"}
+    await hass.services.async_call(LOCK_DOMAIN, SERVICE_UNLOCK, data, blocking=True)
+    await hass.async_block_till_done()
```

Comment:
```suggestion
```
Not needed because blocking=True

Reply:
- [x] yale - b076c08b0d9
- [ ] august -